services/worker/src/worker/job_runners/split/descriptive_statistics.py

```diff
@@ -656,29 +672,35 @@ class AudioColumn(MediaColumn):
     transform_column = FloatColumn
 
     @staticmethod
-    def get_duration(example: dict[str, Any]) -> float:
+    def get_duration(example: Optional[Union[bytes, dict[str, Any]]], is_struct: bool = True) -> Optional[float]:
         """Get audio durations"""
-        with io.BytesIO(example["bytes"]) as f:
+        if example is None:
+            return None
+        example_bytes = example["bytes"] if is_struct else example  # type: ignore
+        with io.BytesIO(example_bytes) as f:
             return librosa.get_duration(path=f)  # type: ignore   # expects PathLike but BytesIO also works
 
     @classmethod
-    def transform(cls, example: dict[str, Any]) -> float:
-        return cls.get_duration(example)
+    def transform(cls, example: dict[str, Any], is_struct: bool = True) -> Optional[float]:
+        return cls.get_duration(example, is_struct)
 
 
 class ImageColumn(MediaColumn):
     transform_column = IntColumn
 
     @staticmethod
-    def get_width(example: dict[str, Any]) -> int:
+    def get_width(example: Optional[Union[bytes, dict[str, Any]]], is_struct: bool = True) -> Optional[int]:
         """Get image widths."""
-        with io.BytesIO(example["bytes"]) as f:
+        if example is None:
+            return None
+        example_bytes = example["bytes"] if is_struct else example  # type: ignore
```

Comment:
`type: ignore` here because example can be both bytes and dict and mypy is not happy with bytes being indexed with string. 
any better solution to this?